File: node/1237.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

/**
 * Escreva a sua solução aqui
 * Code your solution here
 * Escriba su solución aquí
 */

while (lines.length) {
    let word1 = lines.shift()
    let word2 = lines.shift()

    if (!word1 || !word2) {
        break
    }

    word1 = word1.split("")
    word2 = word2.split("")

    let grid = []

    word2.forEach((letter2, rowIndex) => {
        grid.push([])

        word1.forEach((letter1, colIndex) => {
            let value = 0
            if (letter1 != letter2) {
                grid[rowIndex].push(value)
                return
            }

            value = 1

            if (rowIndex === 0 || colIndex === 0) {
                grid[rowIndex].push(value)
                return
            }

            value += grid[rowIndex - 1][colIndex - 1]
            grid[rowIndex].push(value)
        })
    })

    let maxValue = 0
    grid.forEach((row) => {
        row.forEach((col) => {
            if (col > maxValue) {
                maxValue = col
            }
        })
    })

    console.log(maxValue)
    
}

File: python/1237.py

while True: 
    try:
        word1 = input()
        word2 = input()

        grid = []

        for row_index, letter2 in enumerate(word2): 
            grid.append([])
            for col_index, letter1 in enumerate(word1):
                value = 0
                if letter1 != letter2:
                    grid[row_index].append(value)
                    continue

                value = 1

                if row_index == 0 or col_index == 0:
                    grid[row_index].append(value)
                    continue

                value += grid[row_index - 1][col_index - 1]
                grid[row_index].append(value)
            
        max_value = 0
        for row in grid:
            for col_value in row:
                if col_value > max_value:
                    max_value = col_value
        
        print(max_value)

    except:
        break
